info.yml

```diff
@@ -200,3 +200,4 @@ supported_engines:
 
 frameworks:
     - {"name": "tk-framework-shotgunutils", "version": "v5.x.x", "minimum_version": "v5.1.0"}
+    - {"name": "tk-framework-adobe", "version": "v1.x.x"}
```

Comment:
I looked at the `init_framework()` for the adobe framework and it looks like the framework automatically installs the extension at startup. I don't think this is the right pattern, initializing the framework may happen in all sorts of places - it's unexpected. Instead, I would suggest exposing a framework method that can be called - and this method is called from the engine. 

I think it's the engine's responsibility to ensure that the DCC is ready to launch and to have all the necessary pieces installed and prepared - see how it's done in the [photoshop engine](https://github.com/shotgunsoftware/tk-photoshopcc/blob/master/startup.py#L77) today. 

We have this API which engine should implement which encapsulates the launching of a DCC - for more info, see https://developer.shotgunsoftware.com/tk-core/initializing.html#launching-software

Reply:
Hi @manneohrstrom ,
You are right. I migrated the install command into the bootstrap  of the two engines, but currently I'm still using the launchapp to actually get an instance of the framework. The reason is:
If I want to use methods from the framework, I need to have an instance of it somewhere. As the engines (tk-photoshopcc and tk-aftereffectscc) are started from the CEP-panel, the framework would not be instanciated early enough, so we could either add the framework to the launchapp or to the engine running the launchapp (tk-desktop or tk-shotgun).
Do you have another idea on this?
You can look at the latest state of tk-aftereffectscc and tk-framework-adobe to see what I changed.

Reply:
Thanks for the detailed feedback, i'll take a look now!

Generally speaking, I think from a design point of view, the goal should be that we shouldn't have to make any chances to the launch app in order to create a new engine. These two things should be decoupled.

Reply:
ok i see the problem.... trying to think of options...

Reply:
Ok - some followup here @martinminsel : 

I have given this some thought - long term, we really need the ability to import frameworks into the startup state of the engine - had we had that ability today, this would be easy. I did take a look at that, and it's not straight forward and not our highest priority right now, so unfortunately not actionable right now. However, i definitely don't want to introduce this dependency you are adding here (or a dependency on the desktop or shotgun engine) - this is not the direction we want to go.

I think the right solution for the time being is to move the plugin installation code back into the engines (duplicated in two places) and put a big todo in the code explaining that we need to add tk-core support for framework imports in the startup in order to be able to refactor this into the framework. Given our options, this seems like the right one to do at this time. Hope it seems reasonable and makes sense to you too.